src/rdkafka_queue.c

```diff
@@ -582,6 +607,15 @@ int rd_kafka_q_serve_rkmessages (rd_kafka_q_t *rkq, int timeout_ms,
                         continue;
                 }
 
+                if (unlikely(rko->rko_type == RD_KAFKA_OP_BARRIER)) {
+                        cnt = rd_kafka_purge_outdated_messages(
```

Comment:
Cast back to unsigned int (I think MSVC will warn about these lossy auto casts otherwise)